src/main/java/org/numenta/nupic/network/Layer.java

```diff
@@ -2349,11 +2349,12 @@ public ManualInput call(ManualInput t1) {
 
                 @Override
                 public ManualInput call(ManualInput t1) {
-                    if(t1.getFeedForwardSparseActives() == null || t1.getPreviousPredictiveCells() == null) {
+                    int[] ffActiveCols = t1.getFeedForwardSparseActives();
+                    int[] prevPredictedCols = SDR.cellsAsColumnIndices(t1.getPreviousPredictiveCells(), cellsPerColumn);
+                    if(ffActiveCols == null || t1.getPreviousPredictiveCells() == null) {
                         return t1.anomalyScore(1.0);
                     }
-                    return t1.anomalyScore(anomalyComputer.compute(t1.getFeedForwardSparseActives(), 
-                        SDR.cellsAsColumnIndices(t1.getPreviousPredictiveCells(), cellsPerColumn), 0, 0));
+                    return t1.anomalyScore(an.compute(ffActiveCols, prevPredictedCols, 0, 0));
                 }
             };
         }
```

Comment:
This should be temporary to debug the main issue as this is the opposite of what is done above (by creating temporary variables on the heap (i.e. ```ffActiveCols``` & ```prevPredictedCols```). In general my rule of thumb usually is if called 3 or more times then create a temporary variable locally, but not twice. What do you think?